src/Microsoft.AspNetCore.ResponseCaching/Internal/Helpers.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.ResponseCaching.Internal
+{
+    internal static class Helpers
+    {
+        internal static Task CompletedTask { get; } = Task.FromResult(0);
+
+        internal static ConfiguredTaskAwaitable SupressContext(this Task task)
```

Comment:
Not sure about these helpers. They kinda obscure the default API. Only WebListener uses these. We don't do it anywhere else in the stack.